src/type_checker.cpp

```diff
@@ -110,8 +126,21 @@ void TypeChecker::Visit(RecordVarDeclNode& record_decl) {
   if (env_.ProbeSymbol(record_decl.id)) {
     // TODO: redefinition of 'id'
   } else {
-    auto symbol = std::make_unique<SymbolEntry>(record_decl.id,
-                                                record_decl.type->Clone());
+    // NOTE: record_decl doesn't know its own type, it needs to look up in the
+    // type table to update its type.
+    // struct birth { // RecordDeclNode -> stores type entry in type table
+    //   int date;
+    // };
+    //
+    // struct birth bd1 { .date = 1 }; // RecordVarDeclNode -> search type entry
+    // to update its type.
+    // record_type->id() is "birth" in the above example.
+    auto record_type = dynamic_cast<RecordType*>(record_decl.type.get());
+    auto real_type = env_.LookUpType(
+        MangleToRecordTypeId(record_type->id(), record_type->Clone()));
+    assert(real_type);
```

Comment:
To provide better naming, how about we revise the code as the following:
```suggestion
    auto record_type_id = dynamic_cast<RecordType*>(record_decl.type.get())->id();
    auto record_type = env_.LookUpType(
        MangleToRecordTypeId(record_type_id, record_decl.type->Clone()));
    assert(record_type);
```
> [!NOTE]
> With this change, I think the function `MangleToRecordTypeId` should be renamed again as `MangleRecordTypeId`. :disappointed: